logic.js

```diff
@@ -0,0 +1,60 @@
+$(document).ready(function() {
+
+    var config = {
+        apiKey: "AIzaSyCCj0-3DFWplXqAtuq-8Ppn0PjXKXfOwb4",
+        authDomain: "train-schedule-ae260.firebaseapp.com",
+        databaseURL: "https://train-schedule-ae260.firebaseio.com",
+        projectId: "train-schedule-ae260",
+        storageBucket: "",
+        messagingSenderId: "944574872497"
+    };
+    firebase.initializeApp(config);
+
+    var database = firebase.database();
+    var trainRef = database.ref("/trains");
+
+    $("#addTrain").on("submit", function(event) {
+        event.preventDefault();
+
+        trainRef.push({
+            trainName: event.target.trainName.value.trim(),
+            destination: event.target.destination.value.trim(),
+            firstTrain: moment().format(event.target.firstTrain.value, "hh,mm"),
+            frequency: moment().format(event.target.frequency.value, "hh:mm")
+        });
+
+        event.target.reset();
+    });
+
+trainRef.on("child_added", function(childSnapshot) {
+
+	var tFrequency = childSnapshot.val().frequency;
+    var firstTime = childSnapshot.val().firstTrain;
+    var firstTimeConverted = moment(firstTime, "hh:mm").subtract(1, "days");
+    var currentTime = moment();
```

Comment:
Here you have the currentTime, but you are not using it and keep using moment(). This may cause some inconsistency on what the actual current time is because you have multiple instances of moment(). Use currentTime to do your calculations. For example:

```
currentTime.diff(moment(firstTimeConverted), "minutes");
currentTime.add(tMinutesTillTrain, "minutes");
```